api/meetings/views.py

```diff
@@ -0,0 +1,240 @@
+
+from rest_framework import generics, permissions as drf_permissions
+from rest_framework.exceptions import NotFound
+from django.db.models import Q, Count, Subquery, OuterRef, Case, When, Value, CharField, F
+from django.db.models.functions import Length, Substr, Coalesce
+from django.contrib.contenttypes.models import ContentType
+
+from addons.osfstorage.models import OsfStorageFile
+from api.base import permissions as base_permissions
+from api.base.exceptions import InvalidFilterOperator
+from api.base.filters import ListFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.base.versioning import PrivateVersioning
+from api.meetings.serializers import MeetingSerializer, MeetingSubmissionSerializer
+from api.meetings.permissions import IsPublic
+from api.nodes.views import NodeMixin
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from osf.models import AbstractNode, Conference, Contributor, Tag, PageCounter
+from website import settings
+
+
+class MeetingMixin(object):
+    """Mixin with convenience method get_meeting
+    """
+
+    meeting_lookup_url_kwarg = 'meeting_id'
+
+    def get_meeting(self):
+        meeting = get_object_or_error(
+            Conference,
+            Q(endpoint=self.kwargs[self.meeting_lookup_url_kwarg]),
+            self.request,
+            display_name='meeting',
+        )
+        # caching num_submissions on the Conference object
+        meeting.num_submissions = meeting.submissions.count()
+        meeting.save()
+        return meeting
+
+
+class BaseMeetingView(JSONAPIBaseView, MeetingMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.MEETINGS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model = Conference
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+
+    serializer_class = MeetingSerializer
+    view_category = 'meetings'
+
+
+class MeetingList(BaseMeetingView, generics.ListAPIView, ListFilterMixin):
+
+    view_name = 'meeting-list'
+    ordering = ('-modified', )  # default ordering
+
+    # overrides ListFilterMixin
+    def get_default_queryset(self):
+        conferences = Conference.objects.filter(is_meeting=True)
+        for conference in conferences:
+            # caching num_submissions on the Conference objects
+            conference.num_submissions = conference.submissions.count()
+            conference.save()
```

Comment:
I'm using it for lots of things! Main purpose is for easy sorting on `num_submissions` field and bonus, it's an easy display in the serializer. Since it was already a field on the Conference model, I make sure that field is accurate, whether accessed in the List or Detail view.  Then I just return it top-level in the MeetingSerializer for display, you can sort on it if you wish, and it's an easy way to only return Conferences with more than 5 submissions.

Reply:
Ah, okay. So you're keeping it up to date in the model in case other things might need to use that model, rather than just aggregating the data into the view and saving on the writes? Or do you load the meeting object in some other place that's not using this view's information but does need the count?

My main concern is that it seems like a cache that's never used as a cache outside of the transaction. If there were a condition under which it would never update, such as when the meeting becomes inactive or for 5 minutes after the last read, then the cache is useful as a cache. Otherwise it's potentially either 10 or maybe hundreds of db writes (I'm not positive whether this is pre-pagination data or post, but I suspect post) whenever you get the list view. 

If you totally do, that's cool, I'm just making sure I didn't ask the original question poorly.

Reply:
that's a good point, I don't have to cache it on this field - I was using it because it was there.  I can annotate a similarly named field on the queryset instead of writing it each time, same as I'm doing for these other fields like `author_name`

Reply:
there's so many parts to this one:

return all Conferences that have have at least 5 attached public, non-deleted projects with a tag matching the conference endpoint with a least one osfstorage file attached to it

Reply:
Ooooh, right. Like, I know it was done it a super-hacks way, but I keep trying to forget. Okay, let’s leave as is for now, and we can revisit later if it seems important enough to do so. We should definitely try it on prod data before releasing, though, just in case.

Reply:
I've been trying to figure it out, it's bugging me to death!

Reply:
OK - I have a solution! but it doesn't include the files bit - just conferences with at least five submissions (no further restriction that these submissions have a file).  Was wondering if we could reverse the requirement we discussed here: https://github.com/CenterForOpenScience/osf.io/pull/9025#discussion_r282542479

It was already making the query more complex, but with the new solution, I'm not sure how to work it in.

Reply:
Followup- Requirement has been reversed. Discussed at Meetings standup w/ Kevin present.    Meeting submissions are *not* required to have a file (this matches existing functionality), and is now a future improvement.